### ISSUE TITLE
Autopilot create smaller channels than minchansize

### ISSUE THREAD

**Issue by scissorstail** [lightningnetwork/lnd#6925]
### Background

Autopilot create smaller channels than minchansize.
Is it intended or bug?

Also I searched some issues and pull requests might be related this.
#3165
#986

### Your environment

* lnd 0.15.1-beta 
* Linux umbrel 5.10.103-v8+ ... aarch64 GNU/Linux
* bitcoind

### Steps to reproduce

Set `lnd.conf`
```
[autopilot]
...
autopilot.allocation=1
autopilot.minchansize=1000000
autopilot.maxchansize=1000000
...
```
And Have enough funds.

### Expected behaviour

Autopilot create channels that size of 1000000

### Actual behaviour

Autopilot create channels that size of (1000000 - Opening Cost)

examples)
![image](https://user-images.githubusercontent.com/93466598/190892867-d1af9aa2-5385-4a80-907d-c9d6e8e66618.png)

**Comment by positiveblue** [comment]
I guess that [it was added](it was added) to ensure that the channel can always be opened.  The code check the available funds and decides if one more channel can be opened or not without taking into account the fees. Later, to ensure that we did not run out of funds just counting the capacity but that we can actually pay for the fees we subtract it from the capacity itself.

I agree that it would be better if the account for the fees and did not subtract it form the capacity. We can do that by just ignoring the fees and let some channels fail (we cannot pay for the fees/capacity after having opened the previous ones) or subtract a 105% of the channel capacity overtime we decide that we can open the channel, that buffer should ensure that there will always be enough extra funds to pay the fees.
